src/main/scala/viper/server/vsi/HttpServerInterface.scala

```diff
@@ -0,0 +1,131 @@
+package viper.server.vsi
+
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.{Route}
+
+/** At it's core every HTTP server should implement the routes function that describes the HTTP paths.
+  * */
+trait HttpServerInterface {
+  def routes(): Route
```

Comment:
I think this is not specific enough. What routs are needed for a typical verification server? Can we fix these and require that whoever mixes in this trait (a.k.a. implements this interface, using Java terminology) would have to define the implementation for a concrete list of routs?

Reply:
> I think this is not specific enough.

I agree, it's the polar opposite. 😅 

> Can we fix these and require that whoever mixes in this trait (a.k.a. implements this interface, using Java terminology) would have to define the implementation for a concrete list of routs?

Absolutely, this is precisely what the HttpVerificationServerInterface traits does. I kept this file and traits in to provide some context on my thought process, which was to gradually require or provide certain methods which would then result in HttpVerificationServerInterface . Sorry if this was confusing. 

In a sense, none of these traits are really needed. They could all be merged into the HTTPVerificaitonServerInterface trait (requiring the implementation of some fixed route, while providing the possibility to add custom routes).